### ISSUE TITLE
CREATE-SPECIAL-USE fails with "Unexpected extra arguments to Create"

### ISSUE THREAD

**Issue by MASHtm** [cyrusimap/cyrus-imapd#2183]
On 2.5.11 murder setup CREATE-SPECIAL-USE fails
```
. create INBOX.ctest (USE (\Drafts))
. BAD Unexpected extra arguments to Create
```

Since there is no difference between 2.5 branch and master in cmd_create and parsecreateargs() I assume this is true for master as well

**Comment by MASHtm** [comment]
This seems to be a murder issue again

The frontends proxy the following command to the backend:
```
<1508857021<. CREATE INBOX.ctest(USE (\Drafts) PARTITION part06)
>1508857021>. BAD Unexpected extra arguments to Create
```

**Comment by MASHtm** [comment]
most likely this fixes it in 2.5 and master as well
```
--- imapd.c.orig	2017-10-24 17:05:05.808520352 +0200
+++ imapd.c	2017-10-24 17:05:19.123767382 +0200
@@ -5626,7 +5626,7 @@
 
 		    // special use needs extended support, so pass through extargs
 		    if (specialuse.len) {
-			prot_printf(s_conn->out, "(USE (%s)", buf_cstring(&specialuse));
+			prot_printf(s_conn->out, " (USE (%s)", buf_cstring(&specialuse));
 
 			if (partition) {
 			    prot_printf(s_conn->out, " PARTITION ");
```

**Comment by brong** [comment]
Thanks, I've cherry-picked that to master, 3.0 and 2.5 branches.
Good find.
Bron.


On Wed, 25 Oct 2017, at 02:06, Wolfgang Breyha wrote:
> most likely this fixes it in 2.5 and master as well


> --- imapd.c.orig 2017-10-24 17:05:05.808520352 +0200 +++ imapd.c 2017-10-
> 24 17:05:19.123767382 +0200 @@ -5626,7 +5626,7 @@        // special
> use needs extended support, so pass through extargs       if
> (specialuse.len) { -   prot_printf(s_conn->out, "(USE (%s)",
> buf_cstring(&specialuse)); +   prot_printf(s_conn->out, " (USE (%s)",
> buf_cstring(&specialuse));     if (partition) {        prot_printf(s_conn-
> >out, " PARTITION ");
>> — You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub[1], or mute the
> thread[2].> 

--
  Bron Gondwana, CEO, FastMail Pty Ltd
  brong@fastmailteam.com



Links:

  1. https://github.com/cyrusimap/cyrus-imapd/issues/2183#issuecomment-339022760
